MiniAn/minian_main.py

```diff
@@ -467,7 +470,113 @@ def load_doric_to_xarray(
+    # chk, _ = get_optimal_chk(temps, **minian_parameters.params_get_optimal_chk)
+    # temps = temps.chunk({"frame": 1, "height": -1, "width": -1}).rename("temps")
+
+    # estimate shift 
+    shifts = estimate_motion(concatenated_maxAC, dim = "session").compute().rename("shifts")
+
+    # Apply Shifts
+    transformed = apply_transform(concatenated_maxAC, shifts).compute().rename("temps_shifted")
+    shiftds = xr.merge([concatenated_maxAC, shifts, transformed])
+
+    window    = shiftds["temps_shifted"].isnull().sum("session")
+    window, _ = xr.broadcast(window, shiftds["temps_shifted"])
+
+    # get ROI footprints ('A') of the base (reference) file
+    ref_footprints = get_footprints(ref_filename, ref_ROIs, ref.coords)
+
+    # change the unitIds of the current file footprints to number starting from max UnitId in ref file
+    updated_unit_ids = []
+    ref_unit_id_max = (ref_footprints.coords["unit_id"].values).max() + 1
+    for idx, a in enumerate(A.coords["unit_id"]):
+        updated_unit_ids.append(ref_unit_id_max)
+        ref_unit_id_max = ref_unit_id_max + 1
+    A["unit_id"] = updated_unit_ids
+
+    id_dims =  []
+    merged_footprints  = xr.concat([ref_footprints, A], pd.Index(["session1", "session2"], name="session"))
+
+    # Apply shifts to spatial footprint of each session
+    A_shifted = apply_transform(merged_footprints.chunk(dict(height = -1, width = -1)), shiftds["shifts"])
+
+    def set_window(wnd):
+        return wnd == wnd.min()
+    window = xr.apply_ufunc(
+        set_window,
+        window,
+        input_core_dims=[["height", "width"]],
+        output_core_dims=[["height", "width"]],
+        vectorize=True)
+
+    # Calculate centroids
+    cents = calculate_centroids(A_shifted, window)
+                
+    # Calculate centroid distance
+    dist = calculate_centroid_distance(cents, "session", id_dims)
+
+    # Threshold centroid distances
+    dist_ft = dist[dist["variable", "distance"] < param_dist].copy()
+    dist_ft = group_by_session(dist_ft)
+
+    # Generate mappings
+    mappings = calculate_mapping(dist_ft)
+    mappings_meta = resolve_mapping(mappings)
+    mappings_meta_fill = fill_mapping(mappings_meta, cents)
+    mappings_meta_fill.head()
+
+    # update unitIds of the current file footprints if they have a mapped unitID in ref file
+    for i in range(len(mappings_meta_fill)):
+        if mappings_meta_fill.iloc[i]["group"][0] == ("session1", "session2"):
+            updated_unit_ids = [mappings_meta_fill.iloc[i]["session"]["session1"] if x==mappings_meta_fill.iloc[i]["session"]["session2"] else x for x in updated_unit_ids]
+    A["unit_id"] = updated_unit_ids
+        
+    # return currentFile footprints with updated unitIds
+    return A
+
+def get_footprints(ref_filename, ref_ROIs, dimensions):
+
+    file_ = h5py.File(ref_filename, 'r')
+    reference_ROIs =  np.array(file_.get(ref_ROIs))
+
+    unit_id = np.arange(1, len(reference_ROIs), 1)
```

Comment:
Don't think, it's the correct way to load the roi ids, the ids are saved as attributes

Reply:
This is not the ROI ID that we want to fit to, ROI ID is saved as an attribute in the reference file